crates/xline-client/src/clients/lease.rs

```diff
@@ -88,19 +88,22 @@ impl LeaseClient {
     ///         .await?
     ///         .lease_client();
     ///
-    ///     let resp = client.grant(LeaseGrantRequest::new(60)).await?;
+    ///     let resp = client.grant(60, None).await?;
     ///     println!("lease id: {}", resp.id);
     ///
     ///     Ok(())
     /// }
     /// ```
     #[inline]
-    pub async fn grant(&self, mut request: LeaseGrantRequest) -> Result<LeaseGrantResponse> {
-        if request.inner.id == 0 {
-            request.inner.id = self.id_gen.next();
+    pub async fn grant(&self, ttl: i64, id: Option<i64>) -> Result<LeaseGrantResponse> {
+        let mut id = id.unwrap_or_default();
+        if id == 0 {
+            id = self.id_gen.next();
         }
```

Comment:
We can use `unwrap_or_else` here.

Reply:
if id = `Some(0)`, it needs to be generated as well. `unwrap_or_else` may not deal with that.

Reply:
You are right. I didn't take this case into account.